### DISCUSSION TITLE
How to make it use project's config files

### DISCUSSION THREAD

**Discussion by SharakPL** [sdobreff/vscode-php-resolver#2] [Q&A]
Most projects have config files for php-cs-fixer included like:
- `.php_cs.dist` (depreciated)
- `.php-cs-fixer.dist.php`
etc.

How to make this extension follow project rules instead of any default ones?

**Comment by sdobreff** [comment]
Hey @SharakPL, the extension has that setting "phpResolver.fixerConfigString" which accepts PHP config file:
<img width="686" alt="image" src="https://user-images.githubusercontent.com/5992856/212462457-200202f2-9382-40eb-9b8c-e2bc44a2d0e7.png">

There you can insert your custom config PHP file (that setting has higher priority than the JSON rules).

That could be set per project or globally or modified. 
The extension does not search automatically for files, because the project you are working on could be part of other project or some other mix.

Tell me if that works for you.

**Reply by SharakPL** [reply]
Thanks, I noticed this setting, but it would be much better if it automatically searched for common config files in the root of the workspace and use them as primary source, just like other similar tools do (eslint, stylelint, editorconfig, etc.). This setting should be considered a fallback when no config file is found.

**Comment by sdobreff** [comment]
There is as example of what i mean:
Lets say you are developing WP plugin, you also need bunch of other plugins as well, some libs in vendors (which for whatever reason you want to change, and they also have configs) and your project has to be part of the WP install itself, that will require additional computing power to traverse the tree and check for the config. And also - that file could be named whatever you like which makes thing even harder :)

On the other hand - i got your point, and probably the cases I'm describing are not so common.

Maybe it will be best if the extension has that functionality, and the user could turn it on and off.
OK - let me think about that.

Sorry - probably not the answer you've expected 😉
